pkg/controller/sparkapplication/sparkui_test.go

```diff
@@ -180,55 +180,90 @@ func TestCreateSparkUIIngress(t *testing.T) {
 		},
 	}
 
-	service := SparkService{
-		serviceName: app.GetName() + "-ui-svc",
-		servicePort: 4041,
-	}
-	ingressFormat := "{{$appName}}.ingress.clusterName.com"
+	ingressclass_nginx := "nginx"
 
-	expectedIngress := SparkIngress{
-		ingressName: fmt.Sprintf("%s-ui-ingress", app.GetName()),
-		ingressURL:  app.GetName() + ".ingress.clusterName.com",
-	}
-	fakeClient := fake.NewSimpleClientset()
-	sparkIngress, err := createSparkUIIngress(app, service, ingressFormat, fakeClient)
-	if err != nil {
-		t.Fatal(err)
+	app2 := &v1beta2.SparkApplication{
+		Spec: v1beta2.SparkApplicationSpec{
+			UIConfig: &v1beta2.UIConfig{IngressClass: &ingressclass_nginx},
+		},
 	}
 
-	if sparkIngress.ingressName != expectedIngress.ingressName {
-		t.Errorf("Ingress name wanted %s got %s", expectedIngress.ingressName, sparkIngress.ingressName)
+	// list of tests related to app list
+	type args struct {
```

Comment:
Why do you need a `struct` for a single arg?

Reply:
I find it easier to use a generic args struct that can be extended, instead of building a code per case. For example, it can be easily extended. Tell me if you prefer that i only use a more simple object only adapted to my own test.